miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/RelationFinder.java

```diff
@@ -158,9 +163,13 @@ public SampleDto asDto(DetailedSample model) {
           .flatMap(child -> Stream.concat(Stream.of(child).filter(targetChildClass::isInstance), searchChildren(targetChildClass, child)));
     }
 
-    public static Stream<Library> searchChildrenLibraries(DetailedSample model) {
-      return model.getChildren().stream()
-          .flatMap(child -> Stream.concat(child.getLibraries().stream(), searchChildrenLibraries(child)));
+    public static Stream<Library> searchChildrenLibraries(DetailedSample model, LibraryService libraryService) {
+      try {
+        return Stream.concat(libraryService.listBySampleId(model.getId()).stream(), model.getChildren().stream().flatMap(child -> searchChildrenLibraries(child, libraryService)));
+      } catch (IOException e) {
+         log.error("Failed to get child libraries.", e);
```

Comment:
throw exception as that will be less confusing than returning nothing to both users and developers